### DISCUSSION TITLE
Generating 2048 bit DKIM record

### DISCUSSION THREAD

**Discussion by MrRichThomas** [postalserver/postal#2157] [Help with using Postal] [answered]
Hi, just wondered if I could get some advice. We have a customer who wants us to send emails using their domain or the from address, so we generated SPF and DKIM records as normal and sent them on to them. However they are insisting that we generate a 2048 bit DKIM record.

To achieve this would we have to regenerate the signing.key that lives in /opt/postal/config, and would it break existing DKIM records? Or is the key length configurable somewhere?

We are also unsure if we've caused an issue for ourselves as we recently upgraded from Postal v1 to v2 using the 'Installing on a new server with existing data' method: https://docs.postalserver.io/install/upgrade-to-v2#installing-on-a-new-server-with-existing-data but we didn't copy over the signing.key file, so I'm guessing Postal would have generated a new one...? Our DKIM and SPF is passing on sent mail though, so I'm not sure whether its an issue or not.

Thanks in advance for any help.

**Comment by willpower232** [comment, marked as answer]
The signing key file appears to be unrelated to the existing dkim records (#21) so I'm not sure what its for off the top of my head.

**Reply by MrRichThomas** [reply]
Thanks @willpower232 . So, regarding generating new DKIM records for domains in 2048 bit liength, is there a way I can configure Postal to do that?

**Reply by MrRichThomas** [reply]
Is it currently hardcoded...? Line 93: https://github.com/postalserver/postal/blob/main/app/models/domain.rb

**Reply by willpower232** [reply]
I was literally about to tell you that line haha

That value is probably stored in the database so you could replace that with a 2048 bit private key and I believe the UI would show you what the updated DKIM record would be

**Reply by MrRichThomas** [reply]
Brill, thanks Will, I'll set up a test domain and try updating the database with a bigger private key and report back, thanks!

**Reply by MrRichThomas** [reply]
@willpower232 Yeah that worked a treat, thanks!
